compiler/src/main/kotlin/dev/zacsweers/metro/compiler/fir/generators/ContributionsFirGenerator.kt

```diff
@@ -223,13 +237,19 @@ internal class ContributionsFirGenerator(session: FirSession) :
     if (!owner.hasOrigin(Keys.MetroContributionClassDeclaration)) return emptyList()
     val origin = owner.getContainingClassSymbol() as? FirClassSymbol<*> ?: return emptyList()
     val contributions = findContributions(origin) ?: return emptyList()
+    val scope = owner.name.parseEncodedContributionScope()
```

Comment:
Could we generate the scope annotation onto the contribution somewhere to make this less stringy? Even if it's an internal breadcrumb annotation

Reply:
Will give it a shot :+1:, I was thinking we may not be able to copy over the scope parameter to a new annotation if it's not resolved early enough

Reply:
I think as long as we use the right predicate setup though it should work. We've already read the scope class ID to generate the name so 🤞